Sources/SwiftGodotMacroLibrary/MacroGodot.swift

```diff
@@ -173,7 +173,107 @@ class GodotMacroProcessor {
             
             ctor.append("\tclassInfo.registerMethod (name: \"\(getterName)\", flags: .default, returnValue: \(pinfo), arguments: [], function: \(className).\(getterName))\n")
             ctor.append("\tclassInfo.registerMethod (name: \"\(setterName)\", flags: .default, returnValue: nil, arguments: [\(pinfo)], function: \(className).\(setterName))\n")
-            ctor.append("\tclassInfo.registerProperty (\(pinfo), getter: \"\(getterName)\", setter: \"\(setterName)\")")
+            ctor.append("\tclassInfo.registerProperty (\(pinfo), getter: \"\(getterName)\", setter: \"\(setterName)\")\n")
+        }
+    }
+    
+    func processVariantCollectionVariable(_ varDecl: VariableDeclSyntax) throws {
+        guard hasExportAttribute(varDecl.attributes) else {
+            return
+        }
+        guard let last = varDecl.bindings.last else {
+            throw GodotMacroError.noVariablesFound
+        }
+		
+        guard var type = last.typeAnnotation?.type else {
+            throw GodotMacroError.noTypeFound(varDecl)
+        }
+		
+		guard let elementTypeName = type.variantCollectionElementTypeName else {
+			return
+		}
+		
+		// TODO: remove optionality?
+        if let optSyntax = type.as (OptionalTypeSyntax.self) {
+            type = optSyntax.wrappedType
+        }
```

Comment:
I haven't tested optionals yet, and I'm not sure if they should be allowed for collections, because the collection can just be empty?